pkg/postage/service.go

```diff
@@ -202,22 +191,48 @@ func (ps *service) HandleStampExpiry(id []byte) {
 	}
 }
 
-// SetExpired sets expiry for all non-existing batches.
+// SetExpired removes all expired batches from the stamp issuers.
 func (ps *service) SetExpired() error {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
+
 	for _, issuer := range ps.issuers {
 		exists, err := ps.postageStore.Exists(issuer.ID())
 		if err != nil {
 			ps.logger.Error(err, "set expired: checking if issuer exists", "id", issuer.ID())
 			return err
 		}
-		issuer.SetExpired(!exists)
+		if !exists {
+			err := ps.store.Delete(&StampIssuerItem{Issuer: issuer})
+			if err != nil {
+				ps.logger.Error(err, "set expired: delete stamp data", "id", issuer.ID())
```

Comment:
If the node can continue its operation then this should be on the `Debug`/`Waring` level not `Error`.